### ISSUE TITLE
Implement error states for VBA content types

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#16906]
## User Story or Problem Statement
As a Drupal editor, I want to know when I'm making an error or missing a requirement as I'm making edits.

In #16068, Jordan added error states to the [Figma design](https://www.figma.com/file/aKdT4weNNthDxb3r6MolEh/VBA-Content-Specs?type=design&node-id=208-714&mode=design&t=8XISgY7C3ebKqXVz-0). This ticket is to implement the recommended error states in the design.

## Acceptance Criteria
- [ ] Design is reviewed for error states (Jordan is consulted as needed).
- [ ] Error message when required fields are empty matches design as applicable.
- [ ] Error message for specific fields matches design as applicable.
- [ ] Other error messages match design as applicable.
- [ ] Requires design review
- [ ] Requires accessibility review

**Comment by mmiddaugh** [comment]
From UX sync:
Default error states for required fields are handled by Drupal Core/HTML 5/server side browser + UI/UX of the messages can be determined by the client side browser = we do not have very much control over these

**Comment by mmiddaugh** [comment]
Suggestions/questions 
- VBA Banners > Banner alert > "Dismissible by user": suggest adding a reminder that dismissible banners are recommended only in non-emergency situations
- VBA Facility > Prepare for your visit > Accordion headers: suggest adding "This header should use sentence case" (similar to the title guidance for Spotlight content)
- VBA Facility Service > Service details > Service locations: Suggest more meaningful help text for "Add service location" button (also present in VA Service Regions)
- Service Location paragraph type > Hours > Additional hours options: Rather than "brief explanation of `services` offered outside of facility hours" suggest guidance to make it more clear that the field should be used for information relevant to `hours` of operation or availability 
- Service Region > Primary facility  > VBA facilities in this region: I don't think it is possible to enter an invalid VBA facility since the field uses typeahead functionality to suggest a list of possible matches.(Same field is present in Service Region >VBA facilities in this region)

<details>
<summary> Related screenshots </summary>

![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/464e12f4-de1b-4488-8a1b-906d6fd18ce8)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/0adaf104-d38b-4d7b-8d1e-947187131236)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/570e9b0a-7166-49bd-8326-35b7ef7751e9)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/c30c6554-8d96-4e71-9a41-f685f97e4ae4)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/a13fa8ce-aca6-45b5-b461-019149a16f2a)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/5b0b9dc9-1dc5-4534-9b79-2c428b63cb06)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/55cd3639-30e2-4bf3-9e0f-c2c034c809e0)

</details>

**Comment by mmiddaugh** [comment]
@davidmpickett does a Service region require at least one associated VBA facility?
If so, we'll want to indicate that the "VBA facilities in this region" field field is required.

![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/55411834/6b90c541-aeb9-47de-ae0f-79ce262cd3c7)

**Comment by davidmpickett** [comment]
> @davidmpickett does a Service region require at least one associated VBA facility? If so, we'll want to indicate that the "VBA facilities in this region" field field is required.
> 
> ![image](https://private-user-images.githubusercontent.com/55411834/297907495-6b90c541-aeb9-47de-ae0f-79ce262cd3c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2ODE4ODAsIm5iZiI6MTcwNTY4MTU4MCwicGF0aCI6Ii81NTQxMTgzNC8yOTc5MDc0OTUtNmI5MGM1NDEtYWViOS00N2RlLWFlMGYtNzljZTI2MmNkM2M3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDE2MjYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1YTZjNzBjMjBhMzM1MTFmOTIxYjA3YmVkYTVjOTI1YzFkNWVlOWY3ZWJkYzUyOGI4NjdmMTM5MzNiZGM3YjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PS9XjNYUpLate901AUdZK7oTk6a64e_fzGEcd8zmMr8)

This field is intentionally **not** set to required. There is a use case for Service Regions being saved without any facilities attached to them. Could be during initial set-up, or in preparation before a timed launch. Also because we don't delete content, they would need to be saved in Archive even after all the connections to facilities are severed. These are not the primary use cases, but they are legitimate.

Back in May/June 2023 when we discussed the content specs with the CMS Platform team one of the principles that we establishes was that we shouldn't be overly restrictive with required fields. Basically, fields should only be required if the entity cannot function without them. 

We also have a lot of fields that were marked as required "No -> Yes" meaning that we will have them not required during pilot phase and might lock them during/after national rollout

cc @aklausmeier because this is a good content modelling principle to know

**Comment by davidmpickett** [comment]
> * VBA Facility Service > Service details > Service locations: Suggest more meaningful help text for "Add service location" button (also present in VA Service Regions)

There is a [defect](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16452) preventing us from changing this text.

> * Service Region > Primary facility  > VBA facilities in this region: I don't think it is possible to enter an invalid VBA facility since the field uses typeahead functionality to suggest a list of possible matches.(Same field is present in Service Region >VBA facilities in this region)

It is possible to enter random text here. This error message is for the instance where the user has entered some text, but not selected something from the dropdown.

**Comment by aklausmeier** [comment]
@thejordanwood comments added to Figma file _since we have until end of March to migrate this from CMS to VA Figma space_

**Comment by swirtSJW** [comment]
I'd like to get more specific in the ACs than "Matches Design"  only because some what had been in the design are challenging to implement because of the need to patch core and contrib modules.

**Comment by xiongjaneg** [comment]
Because we have limited time in pre-refinement, I'll pull out annotations from the design and put them in the ACs.

**Comment by xiongjaneg** [comment]
From Steve: how hard did we want to make the validations? Particularly for email?

From Dave: We don't want validation that email is va.gov particularly. We need an engineering spike first about predetermined error messages versus ones we can update.

**Comment by xiongjaneg** [comment]
For phone numbers: Do phone numbers need format validation?

From Michelle: We need to accommodate international numbers.

**Comment by xiongjaneg** [comment]
From Michelle: Are there things we can do to fix the issues for the editors, like parentheses are evil, where we can help the editor.

**Comment by xiongjaneg** [comment]
From Christian: The intent of the error states feedback response was to capture in the design what the current error states are. That helps to identify and fill gaps in error states.

**Comment by xiongjaneg** [comment]
This ticket should capture existing standard error states and what would be new. It should also consider impact on other products.

**Comment by mmiddaugh** [comment]
See also [this comment](https://app.zenhub.com/workspaces/sitewide-facilities-639f5253e4b702a32376339e/issues/gh/department-of-veterans-affairs/va.gov-cms/16901#issuecomment-1900674681) regarding meaningful help text for Local spotlights

**Comment by jilladams** [comment]
From planning: Dave noted the next work here is for him to Drupalize the design proposal from Jordan.

**Comment by omahane** [comment]
@jilladams Steve's question about how hard we want to lock down phone validation is not fully answered. Does someone have some guidance on this?

**Comment by omahane** [comment]
Per @davidmpickett, the relevant precursory ticket appears to block this one: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15691

**Comment by jilladams** [comment]
Noting: Laura did us an audit in #17814. As a result, she's gonna draft / share some proposed guidance with the CMS team. That _might_ resolve #15691 as a no-op, or might not and this ticket might still be blocked by Figuring Things Out. 

I've asked Laura to follow up on that ticket once she talks to CMS.

**Comment by Agile6MSkinner** [comment]
@laflannery @jilladams @davidmpickett @omahane I've seen action on phone number validation that would imply that this has all been working out per Jill's comment on 18-Apr. Are error state recommendations and feasibility sufficiently reviewed? Can we close this?

**Comment by davidmpickett** [comment]
Basic validation and error states are in place for VBA. All the fields Jordan indicated that should have error messages have some error messaging. Exact wording may differ, but that is also not entirely in our control due to the various sources of error messages.

A note for the future - because of the nested nature of Drupal fields, it is best to evaluate this error and validation at a component level rather than at a product level. Laura's phone number epic is a good example of this.

TL;DR - This ticket can be closed. We have spent multiple cycles on this and some of this is not totally in our control.
